livepeer/catalyst-api
# Add Mist API client and scaffold for triggering upload VOD

This is the first part of work related to https://github.com/livepeer/go-livepeer/issues/2537

It includes the following parts:
- Mist API client + unit tests
- Scaffold for the upload VOD function

This is how I tested it (and you can too!).

### Testing Mist client

1. Start catalyst node (Dockerized `livepeer/catalyst` is good enough)
2. Check the mist password hash
```
$ git clone https://github.com/DDVTECH/mistserver.git && cd mistserver
$ source api/mist_api.sh
$ mistCall test test '{"variable_list":true}'
$ cat /tmp/MistBashAuth
"authorize":{"username":"test","password":"f0c5a517021bae2634bad2b2c390bbe9"},
```

The string `f0c5a517021bae2634bad2b2c390bbe9` is your password hash.

3. Cherry pick commit with local test params https://github.com/livepeer/catalyst-api/commit/a849c04819b8cbd9c430dd157e71d97e2e7b4e93
4. Update your password hash in https://github.com/livepeer/catalyst-api/commit/a849c04819b8cbd9c430dd157e71d97e2e7b4e93#diff-3b598d3223d4e9fa913b3706f7dbf60e1674f052264ada27e092fd5bf9cc20baR340
5. Start DMS API app
6. Copy sample video into the Catalyst container
```
docker cp Big_Buck_Bunny_1080_10s_1MB.mp4 catalyst-container:/home/
```
7. Start `TestWorkflow()` https://github.com/livepeer/catalyst-api/commit/a849c04819b8cbd9c430dd157e71d97e2e7b4e93#diff-442464ef69390526d2eb9120b73417beffd1586d1c691456bb3ba2d8b420d054R151

8. You should see a notification of a trigger execution in the DMS API logs.
9. You should see that the output video was created in the catalyst container
```
$ docker exec -it catalyst-container sh
# cd /media/recording
# ls
result.ts
```
